frontend/src/metabase/lib/entity-id/hooks/use-validated-entity-id.ts

```diff
@@ -83,3 +91,7 @@ export const useValidatedEntityId = <
     return { id: null, isLoading: false, isError: true } as const;
   }, [isEntityId, isLoading, isError, entity_ids, id]);
 };
+
+function isNumber(num: unknown): num is number {
+  return typeof num === "number" && Number.isFinite(num) && !Number.isNaN(num);
+}
```

Comment:
PS. `isNumber` also exists in `metabase/lib/types` but it's defined as `typeof value === "number"`